packages/x-components/src/components/__tests__/base-extra-params.spec.ts

```diff
@@ -0,0 +1,154 @@
+        },
+        mocks: {
+          $x: {
+            emit: emitSpy
+          }
+        }
+      }
+    );
+  }
+
+  it('emits the default event when none event is passed as prop', () => {
+    renderExtraParams({
+      values: { warehouse: 1234 }
+    });
+
+    expect(emitSpy).toHaveBeenCalledWith('UserChangedRequestParams', { warehouse: 1234 });
+    expect(emitSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('emits a custom event passed as prop', () => {
+    renderExtraParams({
+      events: ['UserChangedNextQueriesRequestParam', 'UserChangedSearchRequestParam'],
+      values: { catalog: 'catalog', warehouse: 1234 }
+    });
+
+    expect(emitSpy).toHaveBeenCalledWith('UserChangedNextQueriesRequestParam', { warehouse: 1234 });
+    expect(emitSpy).toHaveBeenCalledWith('UserChangedNextQueriesRequestParam', {
+      catalog: 'catalog'
+    });
+    expect(emitSpy).toHaveBeenCalledWith('UserChangedSearchRequestParam', { warehouse: 1234 });
+    expect(emitSpy).toHaveBeenCalledWith('UserChangedSearchRequestParam', { catalog: 'catalog' });
+    expect(emitSpy).toHaveBeenCalledTimes(4);
+  });
+
+  // eslint-disable-next-line max-len
+  it('emits the default event when none event is passed as prop and emit the values of the snippet config', () => {
+    renderExtraParams({
+      template: `
+        <Provider>
+          <BaseExtraParams :values="values" />
+        </Provider>
+      `,
+      components: {
+        Provider
+      },
+      values: {
+        store: 'store'
+      }
+    });
+
+    expect(emitSpy).toHaveBeenCalledWith('UserChangedRequestParams', { warehouse: 1234 });
+    expect(emitSpy).toHaveBeenCalledWith('UserChangedRequestParams', { store: 'store' });
+    expect(emitSpy).toHaveBeenCalledTimes(2);
+  });
+
+  // eslint-disable-next-line max-len
+  it('emits a default event when none event is passed as prop and override the values of the snippet config', () => {
+    renderExtraParams({
+      template: `
+        <Provider>
+          <BaseExtraParams :values="values" />
+        </Provider>
+      `,
+      components: {
+        Provider
+      },
+      values: {
+        warehouse: 'warehouse'
+      }
+    });
+
+    expect(emitSpy).toHaveBeenCalledWith('UserChangedRequestParams', { warehouse: 'warehouse' });
+    expect(emitSpy).toHaveBeenCalledTimes(1);
+  });
+
+  // eslint-disable-next-line max-len
+  it('emits a custom event passed as prop and override the values of the snippet config', () => {
+    renderExtraParams({
+      template: `
+        <Provider>
+          <BaseExtraParams :values="values" :events="events" />
+        </Provider>
+      `,
+      components: {
+        Provider
+      },
+      values: {
+        warehouse: 'warehouse'
+      },
+      events: ['UserChangedRecommendationsRequestParam']
+    });
+
+    expect(emitSpy).toHaveBeenCalledWith('UserChangedRecommendationsRequestParam', {
+      warehouse: 'warehouse'
+    });
+    expect(emitSpy).toHaveBeenCalledTimes(1);
+  });
+});
```

Comment:
I'm missing some tests changing the value and the provided value, and asserting the emitted events.